File: CelulaReport/app/src/main/java/com/example/celulareport/viewmodel/ReportListViewModel.java
package com.example.celulareport.viewmodel;

import android.app.Application;

import com.example.celulareport.db.ReportsRepository;
import com.example.celulareport.db.model.ReportCard;
import com.example.celulareport.db.model.ReportEntity;

import java.util.List;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;

public class ReportListViewModel extends AndroidViewModel {
    private final ReportsRepository mRepository;

    public ReportListViewModel(@NonNull Application application) {
        super(application);
        this.mRepository = ReportsRepository.getInstance(application);
    }

    public LiveData<List<ReportCard>> getReportsByMonth(String month){
        return mRepository.getReportsByMonth(month);
    }

    public void Insert(ReportEntity reportEntity){
        mRepository.Insert(reportEntity);
    }

    public void deleteById(long id){
        mRepository.deleteById(id);
    }
}

File: CelulaReport/app/src/main/java/com/example/celulareport/viewmodel/ReportListViewModel.kt
package com.example.celulareport.viewmodel

import androidx.lifecycle.LiveData
import androidx.lifecycle.ViewModel
import androidx.lifecycle.viewModelScope
import com.example.celulareport.db.model.ReportCard
import com.example.celulareport.db.model.ReportEntity
import com.example.celulareport.db.repositories.ReportsRepository
import kotlinx.coroutines.launch

class ReportListViewModel private constructor(private val repository: ReportsRepository) : ViewModel() {

    fun getReportsByMonth(month: String) = repository.getByMonth(month)

    fun insert(report: ReportEntity) {
        viewModelScope.launch {
            repository.insert(report)
        }
    }

    fun deleteReports(vararg report: ReportEntity) {
        viewModelScope.launch {
            repository.deleteAll(*report)
        }
    }


    companion object{

        @Volatile
        private var instace: ReportListViewModel? = null

        fun getInstance(repository: ReportsRepository):ReportListViewModel{
            return instace?: synchronized(this){
                instace?: ReportListViewModel(repository).also { instace = it }
            }
        }
    }
}